### PR TITLE
tests/udev: change test framework asserts to RuntimeError

### PR DESCRIPTION
**Pull request by bgurney-rh**
Related: stratis-storage/project#105

### CODE REVIEW

**Comment by bgurney-rh** [comment]
Rebased.

File: tests/client-dbus/tests/udev/_loopback.py [hunk 1]
```diff
@@ -25,6 +25,9 @@
 
 _SIZE_OF_DEVICE = 1024 ** 4  # 1 TiB
 
+def _check_tokens(self, tokens):
```
Comment (mulkieran) [inline tests/client-dbus/tests/udev/_loopback.py:29]: Move this into the ```LoopBackDevices``` class; that way you're using more standard object orientation.

File: tests/client-dbus/tests/udev/test_udev.py [hunk 2]
```diff
@@ -218,7 +223,9 @@ def start_service(self):
         if service.poll() is not None:
             raise Exception("Daemon unexpectedly exited with %s" % service.returncode)
```
Comment (mulkieran) [inline tests/client-dbus/tests/udev/test_udev.py:224]: Might as well change the ```raise Exception``` here to ```raise RuntimeError``` also.

File: tests/client-dbus/tests/udev/test_udev.py [hunk 3]
```diff
@@ -63,7 +63,7 @@ def random_string(length):
 def _create_pool(name, devices, *, key_description=None):
     """
     Creates a stratis pool. Tries three times before giving up.
-    Raises an assertion error if it does not succeed after three tries.
+    Raises a runtime error if it does not succeed after three tries.
```
Comment (mulkieran) [inline tests/client-dbus/tests/udev/test_udev.py:66]: ^runtime error^RuntimeError OR, better

```
:raises RuntimeError: if pool is not created after three tries
```

File: tests/client-dbus/tests/udev/test_udev.py [hunk 4]
```diff
@@ -133,7 +133,7 @@ def _wait_for_udev(expected_paths):
     expected. Always get the result of at least 1 Stratis enumeration.
     :param expected_paths: devnodes of paths that should belong to Stratis
     :type expected_paths: list of str
-    :return: None (May assert)
+    :return: None (May exit with a RuntimeError)
```
Comment (mulkieran) [inline tests/client-dbus/tests/udev/test_udev.py:136]: Might as well fix this up to be proper header comment:

```
:return: None
:raises RuntimeError: 
```

File: tests/client-dbus/tests/udev/test_udev.py [hunk 5]
```diff
@@ -170,11 +171,12 @@ def _processes(name):
 
 def _remove_stratis_dm_devices():
     """
-    Remove Stratis device mapper devices, fail with an assertion error if
+    Remove Stratis device mapper devices, fail with a runtime error if
```
Comment (mulkieran) [inline tests/client-dbus/tests/udev/test_udev.py:174]: ^runtime error^RuntimeError OR even better

```
:raises RuntimeError: if some devices are remaining
```

File: tests/client-dbus/tests/udev/_loopback.py (hunk 1 above)
Comment (bgurney-rh) [reply tests/client-dbus/tests/udev/_loopback.py:29]: I was getting pylint errors when I first tried that (which probably means I was missing something):

```
./check.py tests
************* Module tests.udev._loopback
tests/udev/_loopback.py:87: [E0602(undefined-variable), LoopBackDevices.unplug] Undefined variable '_check_tokens'
tests/udev/_loopback.py:101: [E0602(undefined-variable), LoopBackDevices.generate_udev_add_events] Undefined variable '_check_tokens'
tests/udev/_loopback.py:121: [E0602(undefined-variable), LoopBackDevices.hotplug] Undefined variable '_check_tokens'
```

The above results if I try this modification to the file, as it is in the PR now:

```
diff --git a/tests/client-dbus/tests/udev/_loopback.py b/tests/client-dbus/tests/udev/_loopback.py
index 09322699..20e6869a 100644
--- a/tests/client-dbus/tests/udev/_loopback.py
+++ b/tests/client-dbus/tests/udev/_loopback.py
@@ -26,11 +26,6 @@ _LOSETUP_BIN = os.getenv("STRATIS_LOSETUP_BIN", "/usr/sbin/losetup")
 _SIZE_OF_DEVICE = 1024 ** 4  # 1 TiB
 
 
-def _check_tokens(self, tokens):
-    if not all(token in self.devices for token in tokens):
-        raise RuntimeError("No tokens found")
-
-
 class LoopBackDevices:
     """
     Class for creating and managing loop back devices which are needed for
@@ -46,6 +41,10 @@ class LoopBackDevices:
         self.count = 0
         self.devices = {}
 
+    def _check_tokens(self, tokens):
+        if not all(token in self.devices for token in tokens):
+            raise RuntimeError("No tokens found")
+
     def create_devices(self, number):
         """
         Create new loop back devices.
```

Comment (mulkieran) [reply tests/client-dbus/tests/udev/_loopback.py:29]: You need to call it as an instance method: ```self._check_tokens(tokens)```.

Comment (bgurney-rh) [reply tests/client-dbus/tests/udev/_loopback.py:29]: Aha; that works.  I was trying changing each call to `self._check_tokens(self, tokens)`, but that was getting "Too many positional arguments for method call".

Comment (bgurney-rh) [reply tests/client-dbus/tests/udev/_loopback.py]: Resolved.

File: tests/client-dbus/tests/udev/_loopback.py [hunk 6]
```diff
@@ -134,10 +139,11 @@ def device_files(self, tokens):
         :return: The list devices corresponding to the tokens
         :rtype: list of str
         :raises: KeyError if any token is not found
-        :raises: AssertionError if any devnode is None
+        :raises: RuntimeError if any devnode is None
         """
         result = [self.devices[token][0] for token in tokens]
-        assert all([devnode is not None for devnode in result])
+        if not all([devnode is not None for devnode in result]):
```
Comment (mulkieran) [inline tests/client-dbus/tests/udev/_loopback.py:145]: Might as well drop the double negative here, while you're doing the other changes:

```
if any([devnode is None for devnode in result]):
```

File: tests/client-dbus/tests/udev/_loopback.py [hunk 7]
```diff
@@ -41,6 +41,10 @@ def __init__(self):
         self.count = 0
         self.devices = {}
 
+    def _check_tokens(self, tokens):
```
Comment (mulkieran) [inline tests/client-dbus/tests/udev/_loopback.py:44]: Please redistribute the negative and change the message:

```
if any(token not in self.devices for token in tokens):
    raise RuntimeError("One of the specified tokens is unknown to this manager")
```

File: tests/client-dbus/tests/udev/test_udev.py (hunk 2 above)
Comment (bgurney-rh) [reply tests/client-dbus/tests/udev/test_udev.py]: Resolved.

File: tests/client-dbus/tests/udev/test_udev.py (hunk 3 above)
Comment (bgurney-rh) [reply tests/client-dbus/tests/udev/test_udev.py]: Resolved.

File: tests/client-dbus/tests/udev/test_udev.py (hunk 4 above)
Comment (bgurney-rh) [reply tests/client-dbus/tests/udev/test_udev.py]: Resolved.

File: tests/client-dbus/tests/udev/test_udev.py (hunk 5 above)
Comment (bgurney-rh) [reply tests/client-dbus/tests/udev/test_udev.py:175]: Resolved.

**Pull request closed without merging**